Automattic/wp-calypso
# (5P) SPT add JS Functional tests to the plugin

This PR aims to add a JS unit test framework to the SPT Plugin via `@wordpress/scripts`.

However, as Gutenberg Core is considering dropping Enzyme the tests are written in React Testing Library. There is also a custom Jest config to allow us to overide the default settings for our Plugin as required.

All components are now covered by a test. Note that some components do not have their own individual test because they are covered by the parent component as we no longer use `shallow` rendering.

### Please note

* Some `devDependencies` have been introduced. These are entirely necessary and have been agreed with @sirreal (see comments below) as being valid.
* The number of lines of code change figure is high. This is however, due to `package-lock.json` and `@wordpress/scripts` dependency. Don't panic.

#### Testing instructions

* `npm install` within the `apps/full-site-editing` dir
* Stay within the `apps/full-site-editing` dir
* Run `npm run test:unit:watch`
* See some example tests pass.
* Write some more tests to see if they get picked up.
